### PR TITLE
Port 'WebSocket Client' demo to Python

### PR DESCRIPTION
**Pull request by rolandlo**
I followed the Javascript implementation closely. It seems to work fine. I didn't find out how to provoke an error (so that `on_error` gets called).

### CODE REVIEW

File: demos/WebSocket Client/main.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+def on_error(_self, err):
+    print("error")
+    print(err, file=sys.stderr)
+
+
+def on_message(_self, data_type, message):
+    if data_type != Soup.WebsocketDataType.TEXT:
+        return
+    str = message.unref_to_array().decode("utf-8")
```
Comment (theCapypara) [inline demos/WebSocket Client/main.py:43]: Please use a different name for the variable, `str` shadows the `str` type.

**Review by theCapypara — CHANGES_REQUESTED** [review]
LGTM, except for one minor thing. Thanks!

**Comment by rolandlo** [comment]
> BTW @rolandlo you've been added to the Workbench team. You should be able to open branches directly on this repo which is a bit easier for us to checkout.
> 
> Would be great if you could do that for future PRs 🙏

Ok, I will do so in future. I was just used to always work on the fork.

**Comment by rolandlo** [comment]
Thanks for all the reviews @theCapypara @sonnyp. It's really a great way of learning.

**Review by theCapypara — APPROVED** [review]

**Pull request merged**